src/cursor/core_cursor.ts

```diff
@@ -813,10 +776,10 @@ function nextFunction(self: any, callback: Callback) {
 
     // Check if connection is dead and return if not possible to
     // execute a getMore on this connection
-    if (isConnectionDead(self, callback)) return;
+    if (isConnectionDead(self as Cursor, callback)) return;
```

Comment:
should not need to cast, `isConnectionDead` takes in a `CoreCursor`

Reply:
This method uses the pool property on CoreCursor which you said we don't need, is Killed or Dead equivalent?

Reply:
yeah hmm, maybe we don't need this method anymore. It was meant to check synchronously whether there was an active connection to be used, but we use `executeOperation` for everything now which will in turn do server selection. Everything is happening asynchronously now, so we can probably get rid of this method. Can you make a ticket to circle back around to this, and we'll leave the casting for now.